pkg/watch/health.go

```diff
@@ -64,69 +78,181 @@ func MonitorPodHealth(config *preparer.PreparerConfig, logger *logging.Logger, s
 			WithField("inner_err", err).
 			Fatalf("error creating health monitor KV store")
 	}
+	// if GetClient fails it means the certfile/keyfile/cafile were
+	// invalid or did not exist. It makes sense to throw a fatal error
+	client, err := config.GetClient()
+	if err != nil {
+		logger.
+			WithField("inner_err", err).
+			Fatalln("failed to get http client for this preparer")
+	}
 
 	node := config.NodeName
 	pods := []PodWatch{}
-	pods = updateHealthMonitors(store, pods, node, logger)
+	pods = updateHealthMonitors(store, client, pods, node, logger)
 	for {
 		select {
 		case <-time.After(POLL_KV_FOR_PODS):
 			// check if pods have been added or removed
 			// starts monitor routine for new pods
 			// kills monitor routine for removed pods
-			pods = updateHealthMonitors(store, pods, node, logger)
+			pods = updateHealthMonitors(store, client, pods, node, logger)
 		case <-shutdownCh:
 			return
 		}
 	}
 }
 
+// Determines what pods should be running (by checking reality store)
+// Creates new PodWatch for any pod not being monitored and kills
+// PodWatches of pods that have been removed from the reality store
+func updateHealthMonitors(store kp.Store,
+	client *http.Client,
+	watchedPods []PodWatch,
+	node string,
+	logger *logging.Logger) []PodWatch {
+	path := kp.RealityPath(node)
+	reality, _, err := store.ListPods(path)
+	if err != nil {
+		logger.WithField("inner_err", err).Warningln("failed to get pods from reality store")
+	}
+
+	return updatePods(store, client, watchedPods, reality, node, logger)
+}
+
+// compares services being monitored with services that
+// need to be monitored.
+func updatePods(store kp.Store,
+	client *http.Client,
+	current []PodWatch,
+	reality []kp.ManifestResult,
+	node string,
+	logger *logging.Logger) []PodWatch {
+	newCurrent := []PodWatch{}
+	// for pod in current if pod not in reality: kill
+	for _, pod := range current {
+		inReality := false
+		for _, man := range reality {
+			if man.Manifest.Id == pod.manifest.Id {
+				inReality = true
+				break
+			}
+		}
+
+		// if this podwatch is not in the reality store kill its go routine
+		// else add this podwatch to newCurrent
+		if inReality == false {
+			pod.shutdownCh <- true
+		} else {
+			newCurrent = append(newCurrent, pod)
+		}
+	}
+	// for pod in reality if pod not in current: create podwatch and
+	// append to current
+	for _, man := range reality {
+		missing := true
+		for _, pod := range newCurrent {
+			if man.Manifest.Id == pod.manifest.Id {
+				missing = false
+				break
+			}
+		}
+
+		// if a manifest is in reality but not current a podwatch is created
+		// with that manifest and added to newCurrent
+		if missing && man.Manifest.StatusPort != 0 {
+			newPod := PodWatch{
+				manifest: man.Manifest,
+				store:    store,
+				statusChecker: StatusChecker{
+					ID:     man.Manifest.Id,
+					Node:   node,
+					URI:    fmt.Sprintf("%s:%d", node, man.Manifest.StatusPort),
```

Comment:
I strongly suggest that you include the full URI here, e.g.: "http://foo.square:1212/_status". This will help remove even more redundant state and code.